lib/AST/ASTPrinter.cpp

```diff
@@ -5065,6 +5115,66 @@ void PrintAST::visitErasureExpr(ErasureExpr *expr) {
 }
 
 void PrintAST::visitKeyPathExpr(KeyPathExpr *expr) {
+  if (expr->isObjC()) {
+    Printer << "#keyPath(";
+  }
+  
+  if (auto root = expr->getRootType()) {
```

Comment:
Methods like this will crash if `isObjC`, writing tests will help catch these issues :)

Reply:
You mean to say I need to `getRootType()` only if it is `!isObjC()`?

Reply:
Yes, for `isObjC` you only need to print the components